test/validation/tests/validation.go

```diff
@@ -0,0 +1,143 @@
+			Expect(err).ToNot(HaveOccurred())
+
+			Expect(len(pods.Items)).To(Equal(1))
+			Expect(pods.Items[0].Status.Phase).To(Equal(corev1.PodRunning))
+		})
+
+		It("Should have the sriov CRDs available in the cluster", func() {
+			crd := &apiext.CustomResourceDefinition{}
+			err := clients.Client.Get(context.TODO(), goclient.ObjectKey{Name: sriovNetworkNodePolicies}, crd)
+			Expect(err).ToNot(HaveOccurred())
+
+			err = clients.Client.Get(context.TODO(), goclient.ObjectKey{Name: sriovNetworkNodeStates}, crd)
+			Expect(err).ToNot(HaveOccurred())
+
+			err = clients.Client.Get(context.TODO(), goclient.ObjectKey{Name: sriovNetworks}, crd)
+			Expect(err).ToNot(HaveOccurred())
+
+			err = clients.Client.Get(context.TODO(), goclient.ObjectKey{Name: sriovOperatorConfigs}, crd)
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+		It("should deploy the injector pod if requested", func() {
+			operatorConfig := &sriovv1.SriovOperatorConfig{}
+			err := clients.Client.Get(context.TODO(), goclient.ObjectKey{Name: "default", Namespace: operatorNamespace}, operatorConfig)
+			Expect(err).ToNot(HaveOccurred())
+
+			if *operatorConfig.Spec.EnableInjector {
+				daemonset, err := clients.DaemonSets(operatorNamespace).Get(context.Background(), "network-resources-injector", metav1.GetOptions{})
+				Expect(err).ToNot(HaveOccurred())
+				Expect(daemonset.Status.DesiredNumberScheduled).To(Equal(daemonset.Status.NumberReady))
+			} else {
+				_, err := clients.DaemonSets(operatorNamespace).Get(context.Background(), "network-resources-injector", metav1.GetOptions{})
+				Expect(err).To(HaveOccurred())
+				Expect(errors.IsNotFound(err)).To(BeTrue())
+			}
+		})
+
+		It("should deploy the operator webhook if requested", func() {
+			operatorConfig := &sriovv1.SriovOperatorConfig{}
+			err := clients.Get(context.TODO(), goclient.ObjectKey{Name: "default", Namespace: operatorNamespace}, operatorConfig)
+			Expect(err).ToNot(HaveOccurred())
+
+			if *operatorConfig.Spec.EnableOperatorWebhook {
+				daemonset, err := clients.DaemonSets(operatorNamespace).Get(context.Background(), "operator-webhook", metav1.GetOptions{})
+				Expect(err).ToNot(HaveOccurred())
+				Expect(daemonset.Status.DesiredNumberScheduled).To(Equal(daemonset.Status.NumberReady))
+			} else {
+				_, err := clients.DaemonSets(operatorNamespace).Get(context.Background(), "operator-webhook", metav1.GetOptions{})
+				Expect(err).To(HaveOccurred())
+				Expect(errors.IsNotFound(err)).To(BeTrue())
+			}
+		})
+
+		It("should have SR-IOV node statuses not in progress", func() {
+			CheckStable()
+		})
+	})
+})
+
+func CheckStable() {
+	res, err := cluster.SriovStable(operatorNamespace, clients)
+	Expect(err).ToNot(HaveOccurred())
+	Expect(res).To(BeTrue(), "SR-IOV status is not stable")
+
+	isClusterReady, err := cluster.IsClusterStable(clients)
+	Expect(err).ToNot(HaveOccurred())
+	Expect(isClusterReady).To(BeFalse(), "Cluster is not stable")
```

Comment:
We shall expect `isClusterReady` to be true here, right?

Reply:
Oh dang, good catch! It's a leftover from me validating the script. Will change

Reply:
Done